graphstorm-processing/scripts/run_repartitioning.py

```diff
@@ -0,0 +1,174 @@
+    parser = script_utils.get_common_parser()  # type: argparse.ArgumentParser
+
+    return parser.parse_args()
+
+
+def main():
+    """Main entry point for re-partitioning."""
+    args = parse_args()
+    logging.basicConfig(level=args.host_log_level)
+    # Remove trailing slash from s3_input_prefix
+    raw_prefix = args.s3_input_prefix
+    s3_input_prefix = raw_prefix[:-1] if raw_prefix[-1] == "/" else raw_prefix
+    bucket = s3_input_prefix.split("/")[2]
+    key_prefix = s3_input_prefix.split("/", maxsplit=3)[3]
+    region = script_utils.get_bucket_region(bucket) if not args.region else args.region
+
+    container_args = [
+        "--input-prefix",
+        s3_input_prefix,
+        "--metadata-file-name",
+        args.config_filename,
+        "--log-level",
+        args.container_log_level,
+    ]
+
+    if args.job_name is None:
+        args.job_name = "gsprocessing-repartitioning"
+
+    timestamp = strftime("%Y-%m-%d-%H-%M-%S", gmtime())
+    processing_job_name = f"{args.job_name}-{timestamp}"
+    if len(processing_job_name) > 63:
+        # Timestamp will be 19 chars, so we take 43 chars from the prefix plus hyphen
+        processing_job_name = f"{args.job_name[:43]}-{timestamp}"
+
+    logging.info("Creating repartitioning job for processed data under %s", s3_input_prefix)
+
+    # Create SageMaker and S3 clients with high retry count to avoid thresholding errors
+    if not region:
+        region = boto3.session.Session().region_name
+    sm_boto, s3_boto = [
+        boto3.client(
+            service,
+            config=Config(
+                region_name=region, connect_timeout=5, read_timeout=60, retries={"max_attempts": 20}
+            ),
+            region_name=region,
+        )
+        for service in ["sagemaker", "s3"]
+    ]
+    sagemaker_session = sagemaker.Session(
+        boto_session=boto3.Session(region_name=region), sagemaker_client=sm_boto
+    )
+
+    processor_kwargs = script_utils.parse_processor_kwargs(args.sm_processor_parameters)
+
+    # Instance configuration
+    byte_size_on_s3 = script_utils.determine_byte_size_on_s3(bucket, key_prefix, s3_boto)
+    input_total_size_in_gb = max(1, byte_size_on_s3 // (1024 * 1024 * 1024))
+    max_allowed_volume_size = script_utils.get_max_volume_size_for_processing(region)
+    # Heuristic, we assume data_size x 2 should suffice for processing
+    if "volume_size_in_gb" not in processor_kwargs:
+        desired_volume_size = 2 * input_total_size_in_gb
+    else:
+        desired_volume_size = processor_kwargs["volume_size_in_gb"]
+        processor_kwargs.pop("volume_size_in_gb")
+    if desired_volume_size > max_allowed_volume_size:
+        logging.warning(
+            "Desired volume size (%d GB) is larger than max required, assigning the max: %d GB",
+            desired_volume_size,
+            max_allowed_volume_size,
+        )
+    required_volume_size_gb = max(min(desired_volume_size, max_allowed_volume_size), 30)
```

Comment:
Why max is 30?

Reply:
I guess the use of max(min) is confusing here:

Given `desired_volume_size=100`, `max_allowed_volume_size=1024`

`min(desired_volume_size, max_allowed_volume_size) = min(100, 1024) = 100` then we take

`max(100, 30)=100`

I'll break this up into two statements to make it more clear.

The `max(x, 30)` ensures that nodes are provided with _at least_ 30GB of storage.